netsync/manager.go

```diff
@@ -427,24 +431,38 @@ func (sm *SyncManager) current() bool {
 // restartSyncIfNeeded finds a new sync candidate if we're not expecting any
 // blocks from the current one.
 func (sm *SyncManager) restartSyncIfNeeded() {
-	if sm.syncPeer != nil {
-		syncPeerState, exists := sm.peerStates[sm.syncPeer]
-		if exists {
-			isWaitingForBlocks := func() bool {
-				syncPeerState.requestQueueMtx.Lock()
-				defer syncPeerState.requestQueueMtx.Unlock()
-				return len(syncPeerState.requestedBlocks) != 0 || len(syncPeerState.requestQueues[wire.InvTypeSyncBlock].queue) != 0
-			}()
-			if isWaitingForBlocks {
-				return
-			}
-		}
+	sm.syncPeerLock.Lock()
+	defer sm.syncPeerLock.Unlock()
+	shouldReplaceSyncPeer, err := sm.shouldReplaceSyncPeer()
+	if err != nil {
+		panic(err)
```

Comment:
I'm grinding my teeth here, but I guess fine, if syncPeer doesn't have peerState, it's indeed a grievous problem that should never happen.

However, in that case maybe move the panic closer to where that happens?